src/FluentValidation/AbstractValidator.cs

```diff
@@ -259,8 +261,12 @@ ValueTask<ValidationResult> completedValueTask
 	}
 
 	private void SetExecutedRuleSets(ValidationResult result, ValidationContext<T> context) {
-		var executed = context.RootContextData.GetOrAdd("_FV_RuleSetsExecuted", () => new HashSet<string>{RulesetValidatorSelector.DefaultRuleSetName});
-		result.RuleSetsExecuted = executed.ToArray();
+				if ((context.RootContextData.TryGetValue("_FV_RuleSetsExecuted", out var obj)) && (obj is HashSet<string> set))

```

Comment:
Bug: The logic here isn't right - it doesn't do the same thing as the previous logic, which leads to test failures. 

Style: Also please could you fix the indentation so it's consistent with the rest of the file?

Reply:
Resolved in 7d91ccb129bd82f82a041002fa14e4490f3a0f33